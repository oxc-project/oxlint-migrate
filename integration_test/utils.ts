import { expect } from 'vitest';
import main from '../src/index.js';
import { OxlintConfig } from '../src/types.js';

export const getSnapshotResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig?: OxlintConfig
) => {
  const collector: string[] = [];
  const result = await main(config, oxlintConfig, {
    reporter: collector.push.bind(collector),
    upgrade: oxlintConfig !== undefined,
  });

  return {
    config: result,
    warnings: collector
      // filter out unsupported rules
      .filter((error) => !error.startsWith('unsupported rule: local/'))
      .filter((error) => !error.startsWith('unsupported rule: perfectionist/'))
      .filter((error) => !error.startsWith('unsupported rule: vue/'))
      .filter((error) => !error.startsWith('unsupported rule: regexp/'))
      .filter((error) => !error.startsWith('unsupported rule: toml/'))
      .filter((error) => !error.startsWith('unsupported rule: style/')),
  };
};

export const getSnapShotUpgradeResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig: OxlintConfig
) => {
  const result = await getSnapshotResult(config, oxlintConfig);
  const upgradedResult = structuredClone(result);
  const result2 = await getSnapshotResult(config, upgradedResult.config);

  expect(result2).toStrictEqual(result);

  return result2;
};
