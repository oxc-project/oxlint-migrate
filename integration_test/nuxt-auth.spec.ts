import { expect, test } from 'vitest';
// @ts-ignore
import nuxt_auth_test from './projects/nuxt-auth.eslint.config.js';
import { getSnapshotResult, getSnapShotUpgradeResult } from './utils.js';

test('nuxt-auth', async () => {
  // https://github.com/antfu/eslint-config?tab=readme-ov-file#plugins-renaming
  const result = await getSnapshotResult(
    nuxt_auth_test.renamePlugins({
      ts: '@typescript-eslint',
      test: 'vitest',
    })
  );
  expect(result).toMatchSnapshot('nuxt-auth');
});

test('nuxt-auth upgrade', async () => {
  const result = await getSnapShotUpgradeResult(
    nuxt_auth_test.renamePlugins({
      ts: '@typescript-eslint',
      test: 'vitest',
    }),
    {
      categories: {
        correctness: 'error',
        perf: 'error',
      },
    }
  );
  expect(result).toMatchSnapshot('nuxt-auth--upgrade');
});
