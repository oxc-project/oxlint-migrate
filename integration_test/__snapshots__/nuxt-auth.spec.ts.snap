// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nuxt-auth > nuxt-auth 1`] = `
{
  "config": {
    "categories": {
      "correctness": "off",
    },
    "env": {
      "browser": true,
      "builtin": true,
      "commonjs": true,
      "es2024": true,
      "node": true,
      "shared-node-browser": true,
    },
    "globals": {
      "computed": "readonly",
      "defineEmits": "readonly",
      "defineExpose": "readonly",
      "defineProps": "readonly",
      "onMounted": "readonly",
      "onUnmounted": "readonly",
      "reactive": "readonly",
      "ref": "readonly",
      "shallowReactive": "readonly",
      "shallowRef": "readonly",
      "toRef": "readonly",
      "toRefs": "readonly",
      "watch": "readonly",
      "watchEffect": "readonly",
    },
    "ignorePatterns": [
      "**/logs",
      "**/*.log",
      "**/npm-debug.log*",
      "**/yarn-debug.log*",
      "**/yarn-error.log*",
      "**/lerna-debug.log*",
      "**/.pnpm-debug.log*",
      "**/report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json",
      "**/pids",
      "**/*.pid",
      "**/*.seed",
      "**/*.pid.lock",
      "**/lib-cov",
      "**/coverage",
      "**/*.lcov",
      "**/.nyc_output",
      "**/.grunt",
      "**/bower_components",
      "**/.lock-wscript",
      "build/Release",
      "**/node_modules/",
      "**/jspm_packages/",
      "**/web_modules/",
      "**/*.tsbuildinfo",
      "**/.npm",
      "**/.eslintcache",
      "**/.stylelintcache",
      "**/.rpt2_cache/",
      "**/.rts2_cache_cjs/",
      "**/.rts2_cache_es/",
      "**/.rts2_cache_umd/",
      "**/.node_repl_history",
      "**/*.tgz",
      "**/.yarn-integrity",
      "**/.env",
      "**/.env.development.local",
      "**/.env.test.local",
      "**/.env.production.local",
      "**/.env.local",
      "**/.cache",
      "**/.parcel-cache",
      "**/.next",
      "**/out",
      "**/.nuxt",
      "**/dist",
      "**/.cache/",
      ".vuepress/dist",
      "**/.temp",
      "**/.cache",
      "**/.docusaurus",
      "**/.serverless/",
      "**/.fusebox/",
      "**/.dynamodb/",
      "**/.tern-port",
      "**/.vscode-test",
      "**/.idea",
      ".yarn/cache",
      ".yarn/unplugged",
      ".yarn/build-state.yml",
      ".yarn/install-state.gz",
      "**/.pnp.*",
      "**/node_modules",
      "**/dist",
      "**/package-lock.json",
      "**/yarn.lock",
      "**/pnpm-lock.yaml",
      "**/bun.lockb",
      "**/output",
      "**/coverage",
      "**/temp",
      "**/.temp",
      "**/tmp",
      "**/.tmp",
      "**/.history",
      "**/.vitepress/cache",
      "**/.nuxt",
      "**/.next",
      "**/.svelte-kit",
      "**/.vercel",
      "**/.changeset",
      "**/.idea",
      "**/.cache",
      "**/.output",
      "**/.vite-inspect",
      "**/.yarn",
      "**/vite.config.*.timestamp-*",
      "**/CHANGELOG*.md",
      "**/*.min.*",
      "**/LICENSE*",
      "**/__snapshots__",
      "**/auto-import?(s).d.ts",
      "**/components.d.ts",
      ".nuxt",
      "**/.nuxt/**",
      ".output",
      "**/.output/**",
      "dist",
      "**/dist/**",
      "node_modules",
      "**/node_modules/**",
      "**/public/**",
    ],
    "overrides": [
      {
        "files": [
          "**/*.?([cm])ts",
          "**/*.?([cm])tsx",
          "**/*.vue",
        ],
        "rules": {
          "constructor-super": "off",
          "getter-return": "off",
          "no-array-constructor": "off",
          "no-class-assign": "off",
          "no-const-assign": "off",
          "no-dupe-class-members": "off",
          "no-dupe-keys": "off",
          "no-func-assign": "off",
          "no-import-assign": "off",
          "no-new-native-nonconstructor": "off",
          "no-obj-calls": "off",
          "no-redeclare": "off",
          "no-setter-return": "off",
          "no-this-before-super": "off",
          "no-undef": "off",
          "no-unreachable": "off",
          "no-unsafe-negation": "off",
          "no-unused-expressions": "off",
          "no-unused-vars": "off",
          "no-useless-constructor": "off",
          "no-var": "error",
          "prefer-rest-params": "error",
          "prefer-spread": "error",
        },
      },
      {
        "files": [
          "**/__tests__/**/*.?([cm])[jt]s?(x)",
          "**/*.spec.?([cm])[jt]s?(x)",
          "**/*.test.?([cm])[jt]s?(x)",
          "**/*.bench.?([cm])[jt]s?(x)",
          "**/*.benchmark.?([cm])[jt]s?(x)",
        ],
        "rules": {
          "no-unused-expressions": "off",
        },
      },
      {
        "files": [
          "**/*.md/**/*.?([cm])[jt]s?(x)",
          "**/*.md/**/*.vue",
        ],
        "rules": {
          "no-alert": "off",
          "no-console": "off",
          "no-labels": "off",
          "no-lone-blocks": "off",
          "no-undef": "off",
          "no-unused-expressions": "off",
          "no-unused-labels": "off",
          "no-unused-vars": "off",
          "unicode-bom": "off",
        },
      },
      {
        "files": [
          "**/scripts/**/*.?([cm])[jt]s?(x)",
        ],
        "rules": {
          "no-console": "off",
        },
      },
      {
        "files": [
          "**/cli/**/*.?([cm])[jt]s?(x)",
          "**/cli.?([cm])[jt]s?(x)",
        ],
        "rules": {
          "no-console": "off",
        },
      },
      {
        "files": [
          "**/*.d.?([cm])ts",
        ],
        "rules": {
          "import/no-duplicates": "off",
        },
      },
      {
        "files": [
          "**/*.config.?([cm])[jt]s?(x)",
          "**/*.config.*.?([cm])[jt]s?(x)",
        ],
        "rules": {
          "no-console": "off",
        },
      },
    ],
    "plugins": [
      "node",
      "jsdoc",
      "import",
      "unicorn",
    ],
    "rules": {
      "array-callback-return": "error",
      "constructor-super": "error",
      "default-case-last": "error",
      "eqeqeq": [
        "error",
        "smart",
      ],
      "import/first": "error",
      "import/no-duplicates": "error",
      "import/no-named-default": "error",
      "import/no-self-import": "error",
      "import/no-webpack-loader-syntax": "error",
      "jsdoc/check-access": "warn",
      "jsdoc/check-property-names": "warn",
      "jsdoc/empty-tags": "warn",
      "jsdoc/implements-on-classes": "warn",
      "jsdoc/no-defaults": "warn",
      "jsdoc/require-param-name": "warn",
      "jsdoc/require-property": "warn",
      "jsdoc/require-property-description": "warn",
      "jsdoc/require-property-name": "warn",
      "jsdoc/require-returns-description": "warn",
      "new-cap": [
        "error",
        {
          "capIsNew": false,
          "newIsCap": true,
          "properties": true,
        },
      ],
      "no-alert": "error",
      "no-array-constructor": "error",
      "no-async-promise-executor": "error",
      "no-caller": "error",
      "no-case-declarations": "error",
      "no-class-assign": "error",
      "no-compare-neg-zero": "error",
      "no-cond-assign": [
        "error",
        "always",
      ],
      "no-console": "off",
      "no-const-assign": "error",
      "no-control-regex": "error",
      "no-debugger": "error",
      "no-delete-var": "error",
      "no-dupe-class-members": "error",
      "no-dupe-keys": "error",
      "no-duplicate-case": "error",
      "no-empty": [
        "error",
        {
          "allowEmptyCatch": true,
        },
      ],
      "no-empty-character-class": "off",
      "no-empty-pattern": "error",
      "no-eval": "error",
      "no-ex-assign": "error",
      "no-extend-native": "error",
      "no-extra-boolean-cast": "error",
      "no-fallthrough": "error",
      "no-func-assign": "error",
      "no-global-assign": "error",
      "no-import-assign": "error",
      "no-invalid-regexp": "off",
      "no-irregular-whitespace": "error",
      "no-iterator": "error",
      "no-labels": [
        "error",
        {
          "allowLoop": false,
          "allowSwitch": false,
        },
      ],
      "no-lone-blocks": "error",
      "no-loss-of-precision": "error",
      "no-multi-str": "error",
      "no-new": "error",
      "no-new-func": "error",
      "no-new-native-nonconstructor": "error",
      "no-new-wrappers": "error",
      "no-obj-calls": "error",
      "no-proto": "error",
      "no-prototype-builtins": "error",
      "no-redeclare": [
        "error",
        {
          "builtinGlobals": false,
        },
      ],
      "no-regex-spaces": "error",
      "no-restricted-globals": [
        "error",
        {
          "message": "Use \`globalThis\` instead.",
          "name": "global",
        },
        {
          "message": "Use \`globalThis\` instead.",
          "name": "self",
        },
      ],
      "no-self-assign": [
        "error",
        {
          "props": true,
        },
      ],
      "no-self-compare": "error",
      "no-shadow-restricted-names": "error",
      "no-sparse-arrays": "error",
      "no-template-curly-in-string": "error",
      "no-this-before-super": "error",
      "no-throw-literal": "error",
      "no-undef": "error",
      "no-unexpected-multiline": "error",
      "no-unreachable": "error",
      "no-unsafe-finally": "error",
      "no-unsafe-negation": "error",
      "no-unused-expressions": [
        "error",
        {
          "allowShortCircuit": true,
          "allowTaggedTemplates": true,
          "allowTernary": true,
        },
      ],
      "no-unused-vars": [
        "error",
        {
          "args": "none",
          "caughtErrors": "none",
          "ignoreRestSiblings": true,
          "vars": "all",
        },
      ],
      "no-useless-call": "error",
      "no-useless-catch": "error",
      "no-useless-constructor": "error",
      "no-useless-rename": "error",
      "no-var": "error",
      "no-with": "error",
      "node/no-exports-assign": "error",
      "node/no-new-require": "error",
      "prefer-exponentiation-operator": "error",
      "prefer-promise-reject-errors": "error",
      "prefer-rest-params": "error",
      "prefer-spread": "error",
      "symbol-description": "error",
      "unicode-bom": [
        "error",
        "never",
      ],
      "unicorn/consistent-empty-array-spread": "error",
      "unicorn/error-message": "error",
      "unicorn/escape-case": "error",
      "unicorn/new-for-builtins": "error",
      "unicorn/no-instanceof-array": "error",
      "unicorn/no-new-array": "error",
      "unicorn/no-new-buffer": "error",
      "unicorn/number-literal-case": "error",
      "unicorn/prefer-dom-node-text-content": "error",
      "unicorn/prefer-includes": "error",
      "unicorn/prefer-node-protocol": "error",
      "unicorn/prefer-number-properties": "error",
      "unicorn/prefer-string-starts-ends-with": "error",
      "unicorn/prefer-type-error": "error",
      "unicorn/throw-new-error": "error",
      "use-isnan": [
        "error",
        {
          "enforceForIndexOf": true,
          "enforceForSwitchCase": true,
        },
      ],
      "valid-typeof": [
        "error",
        {
          "requireStringLiterals": true,
        },
      ],
      "vars-on-top": "error",
      "yoda": [
        "error",
        "never",
      ],
    },
  },
  "warnings": [
    "unsupported rule: accessor-pairs",
    "unsupported rule: antfu/no-top-level-await",
    "unsupported rule: block-scoped-var",
    "unsupported rule: dot-notation",
    "unsupported rule: no-dupe-args",
    "unsupported rule: no-extra-bind",
    "unsupported rule: no-implied-eval",
    "unsupported rule: no-misleading-character-class",
    "unsupported rule: no-octal",
    "unsupported rule: no-octal-escape",
    "unsupported rule: no-restricted-properties",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: no-sequences",
    "unsupported rule: no-undef-init",
    "unsupported rule: no-unmodified-loop-condition",
    "unsupported rule: no-unneeded-ternary",
    "unsupported rule: no-unreachable-loop",
    "unsupported rule: no-use-before-define",
    "unsupported rule: no-useless-backreference",
    "unsupported rule: no-useless-computed-key",
    "unsupported rule: no-useless-return",
    "unsupported rule: object-shorthand",
    "unsupported rule: one-var",
    "unsupported rule: prefer-arrow-callback",
    "unsupported rule: prefer-const",
    "unsupported rule: prefer-regex-literals",
    "unsupported rule: prefer-template",
    "unsupported rule: unused-imports/no-unused-imports",
    "unsupported rule: unused-imports/no-unused-vars",
    "unsupported rule: eslint-comments/no-aggregating-enable",
    "unsupported rule: eslint-comments/no-duplicate-disable",
    "unsupported rule: eslint-comments/no-unlimited-disable",
    "unsupported rule: eslint-comments/no-unused-enable",
    "unsupported rule: node/handle-callback-err",
    "unsupported rule: node/no-deprecated-api",
    "unsupported rule: node/no-path-concat",
    "unsupported rule: node/prefer-global/buffer",
    "unsupported rule: node/prefer-global/process",
    "unsupported rule: node/process-exit-as-throw",
    "unsupported rule: jsdoc/check-param-names",
    "unsupported rule: jsdoc/check-types",
    "unsupported rule: jsdoc/no-multi-asterisks",
    "unsupported rule: jsdoc/require-returns-check",
    "unsupported rule: jsdoc/require-yields-check",
    "unsupported rule: jsdoc/check-alignment",
    "unsupported rule: jsdoc/multiline-blocks",
    "unsupported rule: antfu/import-dedupe",
    "unsupported rule: antfu/no-import-dist",
    "unsupported rule: antfu/no-import-node-modules-by-path",
    "unsupported rule: import/consistent-type-specifier-style",
    "unsupported rule: import/no-mutable-exports",
    "unsupported rule: import/newline-after-import",
    "unsupported rule: command/command",
    "unsupported rule: prefer-const",
    "unsupported rule: ts/ban-ts-comment",
    "unsupported rule: ts/no-array-constructor",
    "unsupported rule: ts/no-duplicate-enum-values",
    "unsupported rule: ts/no-empty-object-type",
    "unsupported rule: ts/no-extra-non-null-assertion",
    "unsupported rule: ts/no-misused-new",
    "unsupported rule: ts/no-namespace",
    "unsupported rule: ts/no-non-null-asserted-nullish-coalescing",
    "unsupported rule: ts/no-non-null-asserted-optional-chain",
    "unsupported rule: ts/no-require-imports",
    "unsupported rule: ts/no-this-alias",
    "unsupported rule: ts/no-unnecessary-type-constraint",
    "unsupported rule: ts/no-unsafe-declaration-merging",
    "unsupported rule: ts/no-unsafe-function-type",
    "unsupported rule: ts/no-unused-expressions",
    "unsupported rule: ts/no-wrapper-object-types",
    "unsupported rule: ts/prefer-as-const",
    "unsupported rule: ts/prefer-literal-enum-member",
    "unsupported rule: ts/prefer-namespace-keyword",
    "unsupported rule: ts/consistent-type-definitions",
    "unsupported rule: ts/consistent-type-imports",
    "unsupported rule: ts/method-signature-style",
    "unsupported rule: ts/no-dupe-class-members",
    "unsupported rule: ts/no-import-type-side-effects",
    "unsupported rule: ts/no-redeclare",
    "unsupported rule: ts/no-use-before-define",
    "unsupported rule: style/array-bracket-spacing",
    "unsupported rule: style/arrow-parens",
    "unsupported rule: style/arrow-spacing",
    "unsupported rule: style/block-spacing",
    "unsupported rule: style/brace-style",
    "unsupported rule: style/comma-dangle",
    "unsupported rule: style/comma-spacing",
    "unsupported rule: style/comma-style",
    "unsupported rule: style/computed-property-spacing",
    "unsupported rule: style/dot-location",
    "unsupported rule: style/eol-last",
    "unsupported rule: style/indent",
    "unsupported rule: style/indent-binary-ops",
    "unsupported rule: style/key-spacing",
    "unsupported rule: style/keyword-spacing",
    "unsupported rule: style/lines-between-class-members",
    "unsupported rule: style/max-statements-per-line",
    "unsupported rule: style/member-delimiter-style",
    "unsupported rule: style/multiline-ternary",
    "unsupported rule: style/new-parens",
    "unsupported rule: style/no-extra-parens",
    "unsupported rule: style/no-floating-decimal",
    "unsupported rule: style/no-mixed-operators",
    "unsupported rule: style/no-mixed-spaces-and-tabs",
    "unsupported rule: style/no-multi-spaces",
    "unsupported rule: style/no-multiple-empty-lines",
    "unsupported rule: style/no-tabs",
    "unsupported rule: style/no-trailing-spaces",
    "unsupported rule: style/no-whitespace-before-property",
    "unsupported rule: style/object-curly-spacing",
    "unsupported rule: style/operator-linebreak",
    "unsupported rule: style/padded-blocks",
    "unsupported rule: style/quote-props",
    "unsupported rule: style/quotes",
    "unsupported rule: style/rest-spread-spacing",
    "unsupported rule: style/semi",
    "unsupported rule: style/semi-spacing",
    "unsupported rule: style/space-before-blocks",
    "unsupported rule: style/space-before-function-paren",
    "unsupported rule: style/space-in-parens",
    "unsupported rule: style/space-infix-ops",
    "unsupported rule: style/space-unary-ops",
    "unsupported rule: style/spaced-comment",
    "unsupported rule: style/template-curly-spacing",
    "unsupported rule: style/template-tag-spacing",
    "unsupported rule: style/type-annotation-spacing",
    "unsupported rule: style/type-generic-spacing",
    "unsupported rule: style/type-named-tuple-spacing",
    "unsupported rule: style/wrap-iife",
    "unsupported rule: style/yield-star-spacing",
    "unsupported rule: style/jsx-closing-bracket-location",
    "unsupported rule: style/jsx-closing-tag-location",
    "unsupported rule: style/jsx-curly-brace-presence",
    "unsupported rule: style/jsx-curly-newline",
    "unsupported rule: style/jsx-curly-spacing",
    "unsupported rule: style/jsx-equals-spacing",
    "unsupported rule: style/jsx-first-prop-new-line",
    "unsupported rule: style/jsx-function-call-newline",
    "unsupported rule: style/jsx-indent-props",
    "unsupported rule: style/jsx-max-props-per-line",
    "unsupported rule: style/jsx-one-expression-per-line",
    "unsupported rule: style/jsx-quotes",
    "unsupported rule: style/jsx-tag-spacing",
    "unsupported rule: style/jsx-wrap-multilines",
    "unsupported rule: antfu/consistent-chaining",
    "unsupported rule: antfu/consistent-list-newline",
    "unsupported rule: antfu/curly",
    "unsupported rule: antfu/if-newline",
    "unsupported rule: antfu/top-level-function",
    "unsupported rule: no-misleading-character-class",
    "unsupported rule: prefer-regex-literals",
    "unsupported rule: test/consistent-test-it",
    "unsupported rule: test/no-identical-title",
    "unsupported rule: test/no-import-node-test",
    "unsupported rule: test/no-only-tests",
    "unsupported rule: test/prefer-hooks-in-order",
    "unsupported rule: test/prefer-lowercase-title",
    "special parser detected: vue-eslint-parser",
    "special parser detected: toml-eslint-parser",
    "ignore list inside overrides is not supported",
    "special parser detected: parser-plain",
    "unsupported rule: curly",
    "unsupported rule: node/prefer-global/process",
    "unsupported plugin for rule: node/no-new-require",
    "unsupported plugin for rule: jsdoc/check-property-names",
    "unsupported plugin for rule: jsdoc/empty-tags",
    "unsupported plugin for rule: jsdoc/implements-on-classes",
    "unsupported plugin for rule: jsdoc/no-defaults",
    "unsupported plugin for rule: jsdoc/require-param-name",
    "unsupported plugin for rule: jsdoc/require-property",
    "unsupported plugin for rule: jsdoc/require-property-description",
    "unsupported plugin for rule: jsdoc/require-property-name",
    "unsupported plugin for rule: jsdoc/require-returns-description",
    "unsupported plugin for rule: import/no-duplicates",
    "unsupported plugin for rule: import/no-named-default",
    "unsupported plugin for rule: import/no-self-import",
    "unsupported plugin for rule: import/no-webpack-loader-syntax",
    "unsupported plugin for rule: unicorn/error-message",
    "unsupported plugin for rule: unicorn/escape-case",
    "unsupported plugin for rule: unicorn/new-for-builtins",
    "unsupported plugin for rule: unicorn/no-instanceof-array",
    "unsupported plugin for rule: unicorn/no-new-array",
    "unsupported plugin for rule: unicorn/no-new-buffer",
    "unsupported plugin for rule: unicorn/number-literal-case",
    "unsupported plugin for rule: unicorn/prefer-dom-node-text-content",
    "unsupported plugin for rule: unicorn/prefer-includes",
    "unsupported plugin for rule: unicorn/prefer-node-protocol",
    "unsupported plugin for rule: unicorn/prefer-number-properties",
    "unsupported plugin for rule: unicorn/prefer-string-starts-ends-with",
    "unsupported plugin for rule: unicorn/prefer-type-error",
    "unsupported plugin for rule: unicorn/throw-new-error",
  ],
}
`;
